torchbenchmark/util/framework/vision/model_factory.py

```diff
@@ -52,14 +57,19 @@ def get_module(self):
         return self.model, self.example_inputs
 
     def train(self, niter=3):
-        optimizer = optim.Adam(self.model.parameters())
-        loss = torch.nn.CrossEntropyLoss()
+        real_input = [ torch.rand_like(self.example_inputs[0]) ]
+        real_output = [ torch.rand_like(self.example_outputs) ]
         for _ in range(niter):
-            optimizer.zero_grad()
-            pred = self.model(*self.example_inputs)
-            y = torch.empty(pred.shape[0], dtype=torch.long, device=self.device).random_(pred.shape[1])
-            loss(pred, y).backward()
-            optimizer.step()
+            self.optimizer.zero_grad()
```

Comment:
Is good to move optimizer step into the loop! Should have been like that all along.